lib/task_listener.js

```diff
@@ -229,7 +228,21 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
-  addRunningTask(runningState) {
+  async recordCapacity () {
```

Comment:
I don't think this or `addRunningTask` or `removeRunningTask` needs to be `async`...
If `lastKnownCapacity` hasn't been set can't we just use `0`. As far as I can see it'll never reach `recordCapacity` without first calling `getTasks` which calls `availableCapacity`...

So if we just set `this.lastKnownCapacity = 0` in the constructor I think we'll do fine. It's not even a bad assumption. If it's working on something and it never has checked for available capacity or attempted to poll for tasks then is it idle? Does it really have any available capacity?

I think it makes sense to initialize it to zero, it avoid an error that would **only** happen if some invariant was violated. Hence, an error that doesn't happen.